freqtrade/optimize/hyperopt_loss/hyperopt_loss_profit_sortino_spring_efficiency.py

```diff
@@ -0,0 +1,139 @@
+    and hybrid drawdown penalty. The loss function is designed to optimize the trading strategies
+    by prioritizing higher profits and lower drawdowns, while adjusting for risk through Sortino
+    ratio.
+    """
+
+    @staticmethod
+    def hyperopt_loss_function(results: DataFrame, trade_count: int,
+                               min_date: datetime, max_date: datetime,
+                               config: Config, *args, **kwargs) -> float:
+        """
+        Calculate the hyperopt loss based on trade results.
+
+        Parameters:
+        - results: DataFrame containing the trading results
+        - trade_count: int representing the number of trades
+        - min_date: datetime object representing the minimum date in the dataset
+        - max_date: datetime object representing the maximum date in the dataset
+        - config: Config object containing the configuration
+
+        Returns:
+        - The calculated loss as a float
+        """
+
+        try:
+            # Total profit
+            profit = results['profit_abs'].sum()
+
+            # Sortino ratio
+            sortino_ratio = calculate_sortino(
+                results, min_date, max_date, config['dry_run_wallet'])
+
+            # Underwater DataFrame (max value to date, drawdown, drawdown relative)
+            drawdown_df = calculate_underwater(
+                results,
+                value_col='profit_abs',
+                starting_balance=config['dry_run_wallet']
+            )
+
+            # Calculate Max Drawdown
+            max_drawdown = abs(min(drawdown_df['drawdown']))
+
+            # Calculate Average Drawdown (absolute and relative)
+            avg_max_drawdown, _ = calculate_average_drawdown(drawdown_df)
+
+            # PSSE Loss Function
+            if profit > 0:
+                # Sortino ratio requires at least 2 trades for meaningful results
+                if trade_count < 2:
+                    sortino_ratio = 1
+
+                # Geometric mean of drawdown penalties
+                geometric_drawdown_penalty = math.sqrt(
+                    (1 + max_drawdown) * (1 + avg_max_drawdown))
+
+                # Harmonic mean of drawdown penalties
+                harmonic_drawdown_penalty = 2 / \
+                    ((1 / (1 + max_drawdown)) + (1 / (1 + avg_max_drawdown)))
+
+                # Hybrid drawdown penalty (geometric mean combined with harmonic mean)
+                hybrid_drawdown_penalty = math.sqrt(
+                    geometric_drawdown_penalty * harmonic_drawdown_penalty)
+
+                # Combine quadratic profit with Sortino ratio and hybrid drawdown penalty
+                loss = -1 * ((profit ** 2) + (profit * sortino_ratio) / hybrid_drawdown_penalty)
+            else:  # profit < 0
+                loss = -profit
+
+            return loss
+
+        # Handle any exception that may occur during loss calculation
+        except Exception:
+            traceback.print_exc()
```

Comment:
```suggestion
            logger.exception()
```
 ?
Though i don't think there should be a generic exception handler in general.

If we expect errors (most likely 0 division errors) - these should be handled explicitly (and silently), not by "let it fail, handle everything identically" - as that will also pollute the output for no good reason.